smtp/mime.ts

```diff
@@ -54,12 +54,12 @@ function encodeBase64(data: Uint8Array) {
 
 	// pad the end with zeros, but make sure to not forget the extra bytes
 	if (extraBytes === 1) {
-		const tmp = data[len - 1];
+		const tmp = data[len - 1] ?? 0;
 		output += LOOKUP[tmp >> 2];
 		output += LOOKUP[(tmp << 4) & 0x3f];
 		output += '==';
 	} else if (extraBytes === 2) {
-		const tmp = (data[len - 2] << 8) + data[len - 1];
+		const tmp = ((data[len - 2] ?? 0) << 8) + (data[len - 1] ?? 0);
```

Comment:
lots of `x ?? 0` might be an opportunity for abstraction to make clear what we are doing. the `tripletToBase64` has a lot to parse before comprehending the bigger picture. 

with that said, i don't have any suggestions in mind just now

Reply:
yeah this is the Deep Dark Code that we took responsibility for when we went zero-dep. it's definitely a cop-out and i very much dislike it as well. since we have the tests it would probably behoove me to come up with a better solution, even probably within this PR (to avoid the scariness of releasing such a cop-out).